extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { ObjectManagementDialogBase, ObjectManagementDialogOptions } from './objectManagementDialogBase';
+import { IObjectManagementService, ObjectManagement } from 'mssql';
+import * as localizedConstants from '../localizedConstants';
+import { CreateDatabaseDocUrl } from '../constants';
+import { DefaultInputWidth } from './dialogBase';
+
+export class DatabaseDialog extends ObjectManagementDialogBase<ObjectManagement.Database, ObjectManagement.DatabaseViewInfo> {
+	private _nameInput: azdata.InputBoxComponent;
+
+	constructor(objectManagementService: IObjectManagementService, options: ObjectManagementDialogOptions) {
+		super(objectManagementService, options);
+	}
+
+	protected override get docUrl(): string {
+		return CreateDatabaseDocUrl;
+	}
+
+	protected override async validateInput(): Promise<string[]> {
+		const errors: string[] = [];
+		if (!this.objectInfo.name) {
+			errors.push(localizedConstants.NameCannotBeEmptyError);
+		} else if (this.viewInfo.databaseNames.some(name => name.toLowerCase() === this.objectInfo.name.toLowerCase())) {
+			errors.push(localizedConstants.DatabaseExistsError(this.objectInfo.name));
+		}
+		return errors;
+	}
+
+	protected async initializeUI(): Promise<void> {
+		let generalSection = this.initializeGeneralSection();
+		let optionsSection = this.initializeOptionsSection();
+		this.formContainer.addItems([generalSection, optionsSection]);
+	}
+
+	private initializeGeneralSection(): azdata.GroupContainer {
+		let containers: azdata.Component[] = [];
+		this._nameInput = this.modelView.modelBuilder.inputBox().withProps({
```

Comment:
please try to use the this.createInput method, so that in the handler you don't need to call the this.runValidation everytime you create a component.